src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts

```diff
@@ -284,6 +284,8 @@ export class ExtensionsViewlet extends ViewContainerViewlet implements IExtensio
 	private defaultRecommendedExtensionsContextKey: IContextKey<boolean>;
 
 	private searchDelayer: ThrottledDelayer<any>;
+	private searchPromises: Map<ViewletPanel, CancelablePromise<any>>;
```

Comment:
I would have map of `panelId` to `Promise` rather than panel.

Reply:
done